Sources/ProjectDescription/FileList.swift

```diff
@@ -2,28 +2,38 @@ import Foundation
 
 public struct FileList: Codable, Equatable {
     /// List glob patterns.
-    public let globs: [Path]
+    public let globs: [FileListGlob]
 
     /// Initializes the files list with the glob patterns.
     ///
-    /// - Parameter globs: Glob patterns.
-    public init(globs: [Path]) {
+    ///   - glob: Relative glob pattern.
+    ///   - excluding: Relative glob patterns for excluded files.
+    public static func list(_ globs: [FileListGlob]) -> FileList {
+        .init(globs)
+    }
+
+    /// Initializes the files list with the glob patterns.
+    ///
+    ///   - glob: Relative glob pattern.
+    ///   - excluding: Relative glob patterns for excluded files.
+    private init(_ globs: [FileListGlob]) {
         self.globs = globs
     }
 
-    public static func == (lhs: FileList, rhs: FileList) -> Bool {
-        lhs.globs == rhs.globs
+    // for backward compatibility when globs property had [Path] type
+    public init(globs: [Path]) {
```

Comment:
Should we mark this deprecated?

Reply:
Yes! I think an annotation shouldn't break the dumping of the manifest as a JSON. Would you mind adding the "deprecated" annotation there @pavel-trafimuk ?

Reply:
Fixed